test/milton/MiltonEvents.t.sol

```diff
@@ -0,0 +1,525 @@
+            address(usdtMockedToken),
+            address(ipTokenUsdt),
+            address(miltonUsdt),
+            address(miltonStorageUsdt),
+            address(stanleyUsdt)
+        );
+        prepareApproveForUsersUsd(_users, usdtMockedToken, address(josephUsdt), address(miltonUsdt));
+        prepareMilton(miltonUsdt, address(josephUsdt), address(stanleyUsdt));
+        prepareJoseph(josephUsdt);
+        prepareIpToken(ipTokenUsdt, address(josephUsdt));
+        // when
+        _miltonSpreadModel.setCalculateQuotePayFixed(TestConstants.PERCENTAGE_4_18DEC); // 4%
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(usdtMockedToken), TestConstants.PERCENTAGE_3_18DEC, block.timestamp); // 3%, PERCENTAGE_3_18DEC
+        vm.prank(_liquidityProvider);
+        josephUsdt.itfProvideLiquidity(TestConstants.USD_28_000_6DEC, block.timestamp);
+        vm.prank(_userTwo);
+        vm.expectEmit(true, true, true, true);
+        emit OpenSwap(
+            1, // swapId
+            _userTwo, // buyer
+            address(usdtMockedToken), // asset
+            MiltonTypes.SwapDirection.PAY_FIXED_RECEIVE_FLOATING, // direction
+            AmmTypes.OpenSwapMoney({
+                totalAmount: TestConstants.USD_10_000_18DEC, // totalAmount
+                collateral: TestConstants.TC_COLLATERAL_18DEC, // collateral
+                notional: 99670098970308907327800, // notional
+                openingFeeLPAmount: TestConstants.TC_OPENING_FEE_18DEC, // openingFeeLPAmount
+                openingFeeTreasuryAmount: TestConstants.ZERO, // openingFeeTreasuryAmount
+                iporPublicationFee: TestConstants.TC_IPOR_PUBLICATION_AMOUNT_18DEC, // iporPublicationFee
+                liquidationDepositAmount: TestConstants.TC_LIQUIDATION_DEPOSIT_AMOUNT_18DEC // liquidationDepositAmount
+            }), // money
+            block.timestamp, // openTimestamp
+            block.timestamp + TestConstants.SWAP_DEFAULT_PERIOD_IN_SECONDS, // endTimestamp, 28 days, PERIOD_28_DAYS_IN_SECONDS
+            MiltonTypes.IporSwapIndicator({
+                iporIndexValue: TestConstants.PERCENTAGE_3_18DEC, // iporIndexValue
+                ibtPrice: 1 * TestConstants.D18, // ibtPrice
+                ibtQuantity: 99670098970308907327800, // ibtQuantity
+                fixedInterestRate: TestConstants.PERCENTAGE_4_18DEC // fixedInterestRate, 4%
+            }) // indicator
+        );
+        miltonUsdt.itfOpenSwapPayFixed(
+            block.timestamp, // openTimestamp
+            TestConstants.USD_10_000_6DEC, // totalAmount
+            TestConstants.PERCENTAGE_6_18DEC, // acceptableFixedInterestRate, 6%
+            TestConstants.LEVERAGE_18DEC // leverage
+        );
+    }
+
+    function testShouldEmitEventWhenOpenReceiveFixedSwap6Decimals() public {
+        // given
+        MockTestnetToken usdtMockedToken = getTokenUsdt();
+        ItfIporOracle iporOracle = getIporOracleAsset(_userOne, address(usdtMockedToken), 0);
```

Comment:
Why you changed EMA to 0 ?? In old test EMA = 3%, please use TestConstants.TC_DEFAULT_EMA_18DEC ! Don't change in new test given params from old test, because you are changing assumption and test will potentially test totaly different behaviour.